include/libcyphal/transport/defines.hpp

```diff
@@ -0,0 +1,90 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef LIBCYPHAL_TRANSPORT_DEFINES_HPP_INCLUDED
```

Comment:
can we call this `types.hpp` instead? We specifically do not want `#define`s

Reply:
I will rename it in the next (347) pr.